Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1

```diff
@@ -0,0 +1,321 @@
+$description = Get-AutomationVariable -Name 'External_AutoStop_Description'
+$timeAggregationOperator = Get-AutomationVariable -Name 'External_AutoStop_TimeAggregationOperator'
+$frequency = Get-AutomationVariable -Name 'External_AutoStop_Frequency'
+$severity = Get-AutomationVariable -Name 'External_AutoStop_Severity'
+$webhookUri = Get-AutomationVariable -Name 'Internal_AutoSnooze_WebhookUri'
+$webhookARMUri = Get-AutomationVariable -Name 'Internal_AutoSnooze_ARM_WebhookUri'
+
+
+try {
+    Write-Output 'Runbook execution started...'
+    $ResourceGroupName = $VMObject.ResourceGroupName
+    $Location = $VMObject.Location
+
+    Write-Output "Location...$($VMObject.Location)"
+
+    Write-Output 'getting status...'
+
+    $NewAlertName = "Alert-$($VMObject.Name)-1"
+
+    if (($VMObject.Type -eq 'Classic') -and ($enableClassicVMs)) {
+        #Processing the alerts for Classy vms
+        #Get the classic VM asset using azure resource graph to get the cloudservice names
+        $AllClassicVMs = Search-AzGraph -Query "Resources | where type =~ 'Microsoft.ClassicCompute/virtualMachines'"
+
+        $currentVM = $AllClassicVMs | Where-Object Name -EQ $VMObject.Name -ErrorAction SilentlyContinue
+
+        $VMState = $currentVM.properties.instanceView.powerState
+
+        $resourceId = "/subscriptions/$($SubId)/resourceGroups/$ResourceGroupName/providers/Microsoft.ClassicCompute/virtualMachines/$($VMObject.Name.Trim())"
+
+        Write-Output "Processing VM ($($VMObject.Name))"
+
+        Write-Output "Current VM state is ($($VMState))"
+
+        $actionWebhook = New-AzAlertRuleWebhook -ServiceUri $WebhookUri
+
+        if ($AlertAction -eq 'Disable') {
+            $ExVMAlerts = Get-AzAlertRule -ResourceGroup $VMObject.ResourceGroupName -DetailedOutput -ErrorAction SilentlyContinue
+
+            if ($ExVMAlerts -ne $null) {
+                Write-Output 'Checking for any previous alert(s)...'
+                #Alerts exists so disable alert
+                foreach ($Alert in $ExVMAlerts) {
+
+                    if ($Alert.Name.ToLower().Contains($($VMObject.Name.ToLower().Trim()))) {
+                        Write-Output "Previous alert ($($Alert.Name)) found and disabling now..."
+                        Add-AzMetricAlertRule -Name $Alert.Name `
+                            -Location $Alert.Location `
+                            -ResourceGroupName $ResourceGroupName `
+                            -TargetResourceId $resourceId `
+                            -MetricName $metricName `
+                            -Operator $condition `
+                            -Threshold $threshold `
+                            -WindowSize $timeWindow `
+                            -TimeAggregationOperator $timeAggregationOperator `
+                            -Action $actionWebhook `
+                            -Description $description -DisableRule
```

Comment:
Put -DisableRule on it's own line

Reply:
Microsoft created this script. We can optimize and fix them in a different PR.